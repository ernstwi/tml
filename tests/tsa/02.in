foo
init bar invar { x <= 10 }

foo -> bar
    guard { x - y == 5 }
    action { press }
bar -> baz
    action { press }
baz -> baz
