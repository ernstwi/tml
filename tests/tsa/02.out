{
    "actions": [
        "press"
    ],
    "clocks": [
        "x",
        "y"
    ],
    "edges": [
        {
            "action": "press",
            "from": "bar",
            "guard": null,
            "reset": null,
            "to": "baz"
        },
        {
            "action": null,
            "from": "baz",
            "guard": null,
            "reset": null,
            "to": "baz"
        },
        {
            "action": "press",
            "from": "foo",
            "guard": [
                {
                    "clock_1": "x",
                    "clock_2": "y",
                    "clocks_n": 2,
                    "cmp": "eq",
                    "n": 5
                }
            ],
            "reset": null,
            "to": "bar"
        }
    ],
    "locations": [
        {
            "id": "bar",
            "initial": true,
            "invariant": [
                {
                    "clock": "x",
                    "cmp": "le",
                    "n": 10
                }
            ]
        },
        {
            "id": "baz",
            "initial": false,
            "invariant": null
        },
        {
            "id": "foo",
            "initial": false,
            "invariant": null
        }
    ]
}
